src/python/pants/backend/python/goals/lockfile.py

```diff
@@ -158,6 +222,140 @@ async def generate_lockfile(
         ),
     )
 
+
+@rule
+async def generate_updated_lockfile(
+    update_req: UpdatePythonLockfileRequest,
+    generate_lockfiles_subsystem: GenerateLockfilesSubsystem,
+    pex_lock_subsystem: PexLockSubsystem,
+    python_setup: PythonSetup,
+) -> ProcessResult:
+    req = update_req.gen_req
+    resolve_config = await Get(ResolvePexConfig, ResolvePexConfigRequest(req.resolve_name))
+
+    for project in pex_lock_subsystem.project:
+        if PipRequirement.parse(project).project_name != project:
+            raise ValueError(
+                f"project {project} is not a bare name; specifier, markers must be specified in Pants targets"
+            )
+
+    # Assuming that any lockfile we are trying to update has a destination
+    # that is a file and not an embedded read only resource.  See
+    # pex_requirements.py
+    maybe_lockfile = await Get(
+        Digest,
+        PathGlobs(
+            [req.lockfile_dest],
+            glob_match_error_behavior=GlobMatchErrorBehavior.ignore,
+        ),
+    )
+    if maybe_lockfile.serialized_bytes_length == 0:
+        raise ValueError(
+            f"Empty or non-existent lockfile at {req.lockfile_dest}. Unable to update in place"
+        )
```

Comment:
Potentially this could be a warning instead? Given the destination doesn't exist, the consequence of generating from scratch inappropriately is just some lost time. 

For my work repo, I'd be leaning towards setting:

```toml
[pex-lock]
update = true
```

in pants.toml, because I don't think I'd ever want to be using "update the whole lockfile" behaviour by default, and it'd be a bit annoying to have to disable it/pass `--no-pex-lock-update` when adding a new resolve.

Reply:
I'm not sure I'm following what the warning would allow. `pex lock update` won't succeed without a lockfile to update.

Reply:
Ah, right. Okay, I guess we'd have to push the decision up a level to get the behaviour I was thinking off (i.e. if the lock doesn't exist go through the other/old code path, rather than this new one).